arcgis-ios-sdk-samples/Route & Directions/Route around barriers/RouteParametersViewController.swift

```diff
@@ -16,44 +16,53 @@
 import UIKit
 import ArcGIS
 
-class RouteParametersViewController: UIViewController {
+class RouteParametersViewController: UITableViewController {
 
-    @IBOutlet var findBestSequenceSwitch: UISwitch!
-    @IBOutlet var preservceFirstStopSwitch: UISwitch!
-    @IBOutlet var preservceLastStopSwitch: UISwitch!
+    @IBOutlet var findBestSequenceSwitch: UISwitch?
+    @IBOutlet var preserveFirstStopSwitch: UISwitch?
+    @IBOutlet var preserveLastStopSwitch: UISwitch?
     
-    var routeParameters: AGSRouteParameters!
+    var routeParameters: AGSRouteParameters? {
+        didSet {
+            setupUI()
+        }
+    }
     
     override func viewDidLoad() {
         super.viewDidLoad()
-
-        self.setupUI()
+        setupUI()
     }
     
-    func setupUI() {
-        if self.routeParameters != nil {
-            self.findBestSequenceSwitch.isOn = self.routeParameters.findBestSequence
-            self.preservceFirstStopSwitch.isOn = self.routeParameters.preserveFirstStop
-            self.preservceLastStopSwitch.isOn = self.routeParameters.preserveLastStop
-            self.enableSubSwitches(self.routeParameters.findBestSequence)
+    private func setupUI() {
+        if let routeParameters = routeParameters {
+            findBestSequenceSwitch?.setOn(routeParameters.findBestSequence, animated: true)
+            
+            preserveFirstStopSwitch?.isEnabled = routeParameters.findBestSequence
+            preserveLastStopSwitch?.isEnabled = routeParameters.findBestSequence
+            
+            if routeParameters.findBestSequence {
+                preserveFirstStopSwitch?.setOn(routeParameters.preserveFirstStop, animated: true)
+                preserveLastStopSwitch?.setOn(routeParameters.preserveLastStop, animated: true)
+            } else {
+                preserveFirstStopSwitch?.setOn(false, animated: true)
+                preserveLastStopSwitch?.setOn(false, animated: true)
+            }
         }
     }
-    
-    func enableSubSwitches(_ enable: Bool) {
-        self.preservceLastStopSwitch.isEnabled = enable
-        self.preservceFirstStopSwitch.isEnabled = enable
-    }
 
     // MARK: - Actions
     
     @IBAction func switchValueChanged(_ sender: UISwitch) {
-        if sender == self.findBestSequenceSwitch {
-            self.routeParameters.findBestSequence = sender.isOn
-            self.enableSubSwitches(sender.isOn)
-        } else if sender == self.preservceFirstStopSwitch {
-            self.routeParameters.preserveFirstStop = self.preservceFirstStopSwitch.isOn
-        } else {
-            self.routeParameters.preserveLastStop = self.preservceLastStopSwitch.isOn
+        switch sender {
+        case findBestSequenceSwitch:
+            routeParameters?.findBestSequence = sender.isOn
+            setupUI()
```

Comment:
Something seems to be off. When both `preserveFirstStop` and `preserveLastStop` are `false`, toggling `findBestSequenceSwitch` toggles both of the other switches. When only one is false, that one stays false when `findBestSequenceSwitch` is toggled. That seems to be a behavior change. Was that intentional?